### DISCUSSION TITLE
ESPHOME Stepper e Display I2c

### DISCUSSION THREAD

**Discussion by MauroStefanelli** [indomus/forum#117] [Hardware & Firmware]
Ciao,
ho costruito delle pompe dosometriche con motori stepper (ULN2003) che gestito con Home Assistant.
Se non visualizzo nulla sul display o se porto il refresh ad un tempo superiore a quanto impelato dagli stepper della pompa tutto funziona se invece visualizzo qualcosa o non metto il tempo di refresh il motore si blocca ogni tanto ( circa 5s).
Da cosa può dipendere?

Grazie mille,
Mauro

`esphome:
  arduino_version: 2.5.1
  name: dosometriche
  platform: ESP8266
  board: nodemcuv2

wifi:
  ssid: "SSID"
  password: "PWD"
  manual_ip:
    static_ip: 192.168.1.229
    gateway: 192.168.1.1
    subnet: 255.255.255.0
  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Dosometriche Fallback Hotspot"
    password: "pwd"

# Display Oled
i2c:
  sda: D1
  scl: D2
  scan: False
  
font:
  - file: 'slkscr.ttf'
    id: font
    size: 10

    
display:
  - platform: ssd1306_i2c
    model: "SSD1306 128x64"
    address: 0x3C
    **update_interval: 60s**
    lambda: |-
      it.print(0, 0, id(font), "Dosometriche");

web_server:
  port: 80
  
captive_portal:

# Enable logging
logger:
  
ota:
  password: "pwd"

      
# Enable Home Assistant API
api:
  password: 'pwd'
  services:
      - service: control_stepper
        variables:
          target: int
        then:
          - stepper.report_position:
              id: dosometrica_pompa_01
              position: 0
          - stepper.set_target:
              id: dosometrica_pompa_01
              target: !lambda 'return target;'

      - service: stepper_speed
        variables:
          target: int
        then:
         - stepper.set_speed:
             id: dosometrica_pompa_01
             speed:  !lambda 'return target;'

      - service: control_stepper_02
        variables:
          target: int
        then:
          - stepper.report_position:
              id: dosometrica_pompa_02
              position: 0
          - stepper.set_target:
              id: dosometrica_pompa_02
              target: !lambda 'return target*-1;'

      - service: stepper_speed
        variables:
          target: int
        then:
         - stepper.set_speed:
             id: dosometrica_pompa_01
             speed:  !lambda 'return target;'
             
      - service: stepper_speed_02
        variables:
          target: int
        then:
         - stepper.set_speed:
             id: dosometrica_pompa_02
             speed:  !lambda 'return target;'
             
stepper:
  - platform: uln2003
    id: dosometrica_pompa_01
    pin_a: D3
    pin_b: D4
    pin_c: GPIO1
    pin_d: GPIO3
    max_speed: 500 steps/s
    sleep_when_done: true    

    acceleration: inf
    deceleration: inf
    
  - platform: uln2003
    id: dosometrica_pompa_02
    pin_a: D5
    pin_b: D6
    pin_c: D7
    pin_d: D8
    max_speed: 500 steps/s
    sleep_when_done: true    

    acceleration: inf
    deceleration: inf`

**Comment by snakuzzo** [comment]
Ti chiedo cortesemente di postare il codice utilizzando il markdown (c'è il tastino apposito nell'editor), altrimenti dovrò cancellare la discussione.
Grazie

**Reply by MauroStefanelli** [reply]
Scusami ma non riesco a trovare il tastino markdown ed ho tolto il codice.

**Reply by snakuzzo** [reply]
Il tastino c'è, basta cercare un minimo... è quello con i simboli "minore" e "maggiore"

![image](https://user-images.githubusercontent.com/7205371/114151895-e92bdf80-991d-11eb-8ecc-f83b8e6ed0cc.png)

poi dai un'occhiata qui...

https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks
